internal/signer/linux/pkcs11/pkcs11.go

```diff
@@ -139,3 +144,24 @@ func (k *Key) Public() crypto.PublicKey {
 func (k *Key) Sign(_ io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
 	return k.signer.Sign(nil, digest, opts)
 }
+
+func (k *Key) EncryptRSA(hash hash.Hash, data []byte) ([]byte, error) {
```

Comment:
Make the public function type independent. Ideally ECP shouldn't have to care what the type of the PKCS #11 key is.

Reply:
I'm a little bit confused about what you mean by type independent. Is it that I shouldn't make it a function within the Key struct? Or is it that I should not have explicit type assertions for RSA keys? I was thinking that I could make the EncryptRSA function private and make a separate public function called Encrypt that checks the key type and calls the appropriate private function.

Reply:
I mean that consumers of your API need to change their code if they switch from using an RSA key to an EC key. Ideally it should require no code changes for them.

Your thinking is correct, remove the burden of type checking from the API caller.